### DISCUSSION TITLE
Implementing hot-reload

### DISCUSSION THREAD

**Discussion by WhoAteDaCake** [mint-lang/mint#478] [Language Development]
Hi, I've started looking into hot-reload for mint to speed up development. This is a long term process for me to better understand bundlers, so I wouldn't expect it to be completed soon. Good news is that preact supports it, bad news is that it requires quite a bit of work:

* Currently mint-runtime uses a precompiled version of preact, meaning [prefresh](https://github.com/preactjs/prefresh) can't inject it's hooks
* Mint compontents are not exported anywhere, meaning prefetch can't compare classes, to find, which updates need to be triggered.
* Mint components have minimized names within compiled javascript code even in non-production environment, making it a bit harder to reference them when checking for updates.

My naive approach for this:
* Update compiler to compare two ASTs to find changes
* Change compiler to generate js components with the same name as they are defined in .mint files
* Update mint-runtime to use module system (only supported in browser browsers) instead of full bundle
* Export all components generated by the compiler 
* Separate render process from generated components file
* Load preact from somewhere like **unpkg**
* Reactor server implements new logic, where it compares new AST vs previous to see changes
  * Send what was updated via websocket
* Livereload: If it's assets reload page, if it's a component or hooks we can do a manual update

Do you think my assumptions are correct and that's general approach to follow ? I'm very open to any feedback on this :)

**Comment by gdotdesign** [comment]
The compiler is so fast that it's pointless to do hot reloading.
